src/scripts/main.js

```diff
@@ -1,3 +1,85 @@
 'use strict';
 
-// write code here
+const refs = {
+  buttons: {
+    appendRow: document.querySelector('.append-row'),
+    removeRow: document.querySelector('.remove-row'),
+    appendColumn: document.querySelector('.append-column'),
+    removeColumn: document.querySelector('.remove-column'),
+  },
+  table: document.querySelector('.field'),
+};
+
+const MAX_ROWS = 10;
+const MIN_ROWS = 2;
+const MAX_COLUMNS = 10;
+const MIN_COLUMNS = 2;
+
+function getRowCount() {
+  return refs.table.rows.length;
+}
+
+function getColumnCount() {
+  return getRowCount() > 0 ? refs.table.rows[0].cells.length : 0;
```

Comment:
Potential error: Accessing `refs.table.rows[0]` without checking if any rows exist can lead to errors if the table is empty. Consider adding a check to ensure `refs.table.rows.length > 0` before accessing `cells.length`.